src/app.js

```diff
@@ -7,6 +7,19 @@ export function app(app) {
   var node
   var element
 
+  function hydrate(elm) {
+    return elm == null ? undefined : {
+      tag: elm.tagName,
+      data: {},
+      children: [].map.call(elm.childNodes, function(child){hydrate(child)})
+    }
+  }
+
+  if (root.hasChildNodes !== undefined && root.hasChildNodes()) {
+    element = root.children[0]
+    node = hydrate(element)
+  }
+
```

Comment:
@ngryman Can I have your input of:

> andyrj Is that the expected behavior, if root is not empty then append to children?

Basically, in order to hydrate the initial undefined node tree, the root must be non-empty. You'll recall that we didn't support that use case until you introduced the feature decades ago.